.github/workflows/codespell.yaml

```
@@ -0,0 +1,24 @@
+name: codespell
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+jobs:
+  codespell:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Codespell with annotations
+        uses: codespell-project/actions-codespell@v1
+        with:
+          builtin: clear,rare,informal
+          check_filenames: true
+          check_hidden: true
+          ignore_words_file: .codespellignore
+          only_warn: true
+          skip: "./.git/*,go.mod,go.sum,./semconv/*"
+          uri_ignore_words_list: "*"
```

How about 
```suggestion
        run: make codespell
```
?

1. Less dependencies (we would not depend on `codespell-project/actions-codespell` action
2. No need to sync the `make codespell` and `codespell` GitHub workflow